### PR TITLE
#issue89: line can be split between unary operator not

### PR DESCRIPTION
**Pull request by felipevolpone**
This pull request refers to the https://github.com/google/yapf/pull/115 pull request. I just changed a little based on the tips of @gwelymernans .

### CODE REVIEW

**Comment by felipevolpone** [comment]
@gwelymernans  Can you please check again?

**Comment by gwelymernans** [comment]
@felipevolpone I'm sorry for the delay. I still don't think this is the correct approach. You're checking for `not` as the previous token and `True` or `False` for the current token. But that won't work for all situations. I don't think you need to check that the current token is anything, only that the previous token is `not`.

File: yapf/yapflib/unwrapped_line.py [hunk 1]
```diff
@@ -383,6 +383,11 @@ def _IsSurroundedByBrackets(tok):
 
 def _SplitPenalty(prev_token, cur_token):
   """Return the penalty for breaking the line before the current token."""
+
+  if prev_token.value == 'not' and (cur_token.value == 'True'
```
Comment (hayd) [inline yapf/yapflib/unwrapped_line.py]: I can confirm that this passes your tests as simply:

    if prev_token.value == 'not':

change this, squash and rebase. I think this is a nice enhancement!

File: yapftests/reformatter_test.py [hunk 2]
```diff
@@ -1914,6 +1914,30 @@ def testExecAsNonKeyword(self):
     self.assertEqual(expected_formatted_code, reformatter.Reformat(uwlines))
 
 
+class TestingNotInParameters(unittest.TestCase):
+
+    def test_notInParams(self):
+        unformatted_code = textwrap.dedent("""\
+            def sum(a, sprint):
```
Comment (hayd) [inline yapftests/reformatter_test.py]: (Though I don't like your test overriding the builtin sum function :p these tests would work just as well with just the last line.... but I don't think this matters.)

**Comment by felipevolpone** [comment]
@gwelymernans  and @hayd: thanks for the help. 
Based on what @hayd said I created another pr #149. If you could see and give some feedback, it would be great.

**Comment by hayd** [comment]
FWIW you can "force push" changes to a branch and it'll update the pull-request (no need to do it this time but perhaps good to know).

**Comment by gwelymernans** [comment]
This seems to be done now with #149. Thanks!

**Pull request closed without merging**